homeassistant/components/__init__.py

```diff
@@ -143,9 +143,11 @@ def async_handle_turn_service(service):
         yield from asyncio.wait(tasks, loop=hass.loop)
 
     hass.services.async_register(
-        ha.DOMAIN, SERVICE_TURN_OFF, async_handle_turn_service)
+        ha.DOMAIN, SERVICE_TURN_OFF, async_handle_turn_service,
+        state_to_set=STATE_OFF)
     hass.services.async_register(
-        ha.DOMAIN, SERVICE_TURN_ON, async_handle_turn_service)
+        ha.DOMAIN, SERVICE_TURN_ON, async_handle_turn_service,
+        state_to_set=STATE_ON)
```

Comment:
That is a very high repeat rate. So we can say that SERVICE_TURN_ON = STATE_ON and SERVICE_TURN_OFF = STATE_OFF

So I think we need a lookup for SERVICE_XY to STATE and not state_to_set parameter